src/components/AddTodoForm/AddTodoForm.tsx

```diff
@@ -0,0 +1,93 @@
+import { useState } from 'react';
+import { User } from '../../types/User';
+
+interface Props {
+  users: User[];
+  addTodo: (title: string, userId: number) => void;
+}
+
+export const AddTodoForm = (props: Props) => {
+  const { users, addTodo } = props;
+
+  const [selectedUser, setSelectedUser] = useState<number | null>(null);
+  const [newTitle, setNewTitle] = useState('');
+
+  const [showErrors, setShowErrors] = useState(false);
+
+  function resetForm() {
+    setSelectedUser(null);
+    setNewTitle('');
+  }
+
+  function handleSubmit(event: React.FormEvent) {
+    event.preventDefault();
+    if (!showErrors) {
+      setShowErrors(true);
+    }
+
+    if (selectedUser && newTitle.trim()) {
+      addTodo(newTitle, selectedUser);
+      resetForm();
+      setShowErrors(false);
+    }
+  }
+
+  return (
+    <>
+      <h1>Add todo form</h1>
+
+      <form
+        action="/api/todos"
+        method="POST"
+        onSubmit={handleSubmit}
+      >
+        <div className="field">
+          <label htmlFor="titleInput">
+            {'Title: '}
+            <input
+              id="titleInput"
+              type="text"
+              data-cy="titleInput"
+              value={newTitle}
+              placeholder="Enter a title"
+              onChange={(event) => setNewTitle(event.target.value)}
+            />
+          </label>
+          {(!newTitle && showErrors) && (
+            <span className="error">Please enter a title</span>
+          )}
+        </div>
+
+        <div className="field">
+          <label htmlFor="userSelect">
+            {'User: '}
+            <select
+              name="userSelect"
+              data-cy="userSelect"
+              value={selectedUser !== null ? selectedUser : 0}
```

Comment:
```suggestion
              value={selectedUser !== null ? selectedUser : defaultUserValue}
```
What does mean zero? Need constant which describes what means this value ( magic numbers )